tests/test_handler_metrics_saver_dist.py

```diff
@@ -79,7 +80,7 @@ def _all_gather(engine):
         metrics_saver.attach(engine)
         engine.run(data, max_epochs=1)
 
-        if dist.get_rank() == 0:
```

Comment:
Hi @wyli ,

Thanks for your quick fix.
I think maybe it's better to add `dist.barrier()` call at end of the test, in case we face other similar "early exit":
https://pytorch.org/docs/stable/distributed.html#torch.distributed.barrier
CC @vfdev-5 

Thanks.

Reply:
Hi @wyli ,

Sorry maybe I didn't make it clear, I mean in case we add other `dist` call except `get_rank()` in future, maybe it's better to make sure the 2 processes exit together.

Thanks.

Reply:
@Nic-Ma in our test setup we call barrier before destroying dist group: https://github.com/pytorch/ignite/blob/192d721db38ae8587742bf49d4e66ab3c66c8596/tests/ignite/conftest.py#L105-L106

Maybe, this help to prevent execution divergences between tests. Also, we are trying to avoid setting up new dist group for another test while previous test is not yet finished by all procs...

Reply:
Hi @vfdev-5 ,

Thanks for your sharing, that's exactly what I also want to mark:
https://github.com/pytorch/ignite/issues/2185#issuecomment-916383533
@wyli What do you think? Maybe it's better to change the `DistCall` test class for teardown part?

Thanks.

Reply:
Thanks for the information! 

In our case the unit tests run independently from the commandline, perhaps we don't have that issue across different tests. 

also to simulate the user behaviour as much as possible, the barrier is not needed at the moment... example usages are here in [our tutorial](https://github.com/Project-MONAI/tutorials/blob/77465cc80122b921956f98b6ec9a6573fffb4539/acceleration/distributed_training/brats_training_ddp.py#L322) or [here from pytorch]( https://github.com/pytorch/examples/blob/master/distributed/ddp/main.py#L20-L21). 

@Nic-Ma  how about we take a note for now, if there are further issues we review the multiprocessing exiting logic?

Reply:
Hi @wyli ,

OK, we can leave it for the future if more related issues come.
Anyway, this PR can at least fix the CI issue.

Thanks.

Reply:
And BTW, maybe we also have similar issue in other dist tests, could you please help double confirm?

Thanks.

Reply:
sure, I'll keep this PR open and trigger the CI a few times to confirm.